enginetest/queries.go

```diff
@@ -1750,6 +1750,60 @@ var QueryTests = []QueryTest{
 		Query:    `SELECT SUBSTRING_INDEX(mytable.s, "d", 1) AS s FROM mytable INNER JOIN othertable ON (SUBSTRING_INDEX(mytable.s, "d", 1) = SUBSTRING_INDEX(othertable.s2, "d", 1)) GROUP BY 1 HAVING s = 'secon'`,
 		Expected: []sql.Row{{"secon"}},
 	},
+
+	{
+		Query:    `SELECT TRIM("   foo   ")`,
+		Expected: []sql.Row{{"foo"}},
+	},
+	{
+		Query:    `SELECT TRIM(" " FROM "   foo   ")`,
+		Expected: []sql.Row{{"foo"}},
+	},
+	{
+		Query:    `SELECT TRIM(LEADING " " FROM "   foo   ")`,
+		Expected: []sql.Row{{"foo   "}},
+	},
+	{
+		Query:    `SELECT TRIM(TRAILING " " FROM "   foo   ")`,
+		Expected: []sql.Row{{"   foo"}},
+	},
+	{
+		Query:    `SELECT TRIM(BOTH " " FROM "   foo   ")`,
+		Expected: []sql.Row{{"foo"}},
+	},
+	{
+		Query:    `SELECT TRIM("" FROM " foo")`,
+		Expected: []sql.Row{{" foo"}},
+	},
+	{
+		Query:    `SELECT TRIM("bar" FROM "barfoobar")`,
+		Expected: []sql.Row{{"foo"}},
+	},
+	{
+		Query:    `SELECT TRIM(TRAILING "bar" FROM "barfoobar")`,
+		Expected: []sql.Row{{"barfoo"}},
+	},
+	{
+		Query:    `SELECT TRIM(TRAILING "foo" FROM "foo")`,
+		Expected: []sql.Row{{""}},
+	},
+	{
+		Query:    `SELECT TRIM(LEADING "ooo" FROM TRIM("oooo"))`,
+		Expected: []sql.Row{{"o"}},
+	},
+	{
+		Query:    `SELECT TRIM(LEADING "test" FROM TRIM("  test  "))`,
+		Expected: []sql.Row{{""}},
+	},
+	{
+		Query:    `SELECT TRIM(LEADING 1 FROM "11111112")`,
+		Expected: []sql.Row{{"2"}},
+	},
+	{
+		Query:    `SELECT TRIM(LEADING 1 FROM 11111112)`,
```

Comment:
You want a few test cases with a match of the trim phrase in the middle of the word that don't get trimmed, e.g. trim(trailing "a" from "baba") == bab, trim(trailing "ab" from "baba") == baba

Reply:
Also want some tests cases where the expressions for trimming are not literals

Some where they are column values, some where they are function calls (e.g. concat("a", "b"))